wiki/multilang_md.py

```diff
@@ -0,0 +1,333 @@
+            suburl = suburl.replace("  ", " ").replace(" ", "-")
+            # toc line
+            temp = f"1. [{head}](#{suburl})\n"
+            # append
+            _toc_str += "    " * (_level - min_)
+            _toc_str += temp
+        # recursive
+        _toc_str = create_toc(
+            child, min_, max_, enable_emoji_header, _toc_str, _level + 1
+        )
+    return _toc_str
+
+
+class CreateMonolangualDoc(object):
+    def __init__(self, path, file_name, lang, suffix=True):
+        self._base_file = file_name
+        fname, base, ename = self._base_file.split(".")
+
+        new_name = fname + ("." + lang + "." if suffix else ".") + ename
+        self.full_name = os.path.join(path, new_name)
+
+        self.doc = open(self.full_name, "w", encoding="utf-8")
+
+        self.content = []
+
+        self.toc = OrderedDict()
+        self.header_re = re.compile("#+ ")
+        self.toc_stack = []
+        self.prev_level = 0
+
+        self.success = False
+
+    def __del__(self):
+        if self.success:
+            self.__save()
+        self.doc.close()
+        if (not self.success) and (os.path.isfile(self.full_name)):
+            os.remove(self.full_name)
+
+    def __repr__(self):
+        return self.full_name
+
+    def __save(self):
+        for line in self.content:
+            if isinstance(line, tuple):  # toc
+                min_, max_, enable_emoji_header = line
+                self.doc.write(create_toc(self.toc, min_, max_, enable_emoji_header))
+            else:
+                self.doc.write(str(line))
+
+    def CustomException(self, message):
+        return click.ClickException(f"<{self._base_file}> {message}")
+
+    def write(self, oneline, codeblock_mark):
+        # write
+        self.content.append(oneline)
+        # check header
+        if not codeblock_mark:
+            result = self.header_re.match(oneline)
+            if result:
+                head = oneline[result.end():].replace("\n", "")
+                self.append_toc(head, result.end() - 1)
+
+    def append_toc(self, head, level):
+        if (self.prev_level + 1) == level:  # new child
+            pass
+        elif self.prev_level == level:  # same level
+            self.toc_stack = self.toc_stack[:-1]
+        elif self.prev_level > level:  # new elem
+            self.toc_stack = self.toc_stack[: (level - 1)]
+        else:
+            msg = "'Table of Contents' level mismatch. prev_level="
+            msg += f"{self.prev_level}, current_level={level}"
+            raise self.CustomException(msg)
+        recursive = self.toc
+        for x in self.toc_stack:
+            recursive = recursive[x]
+        recursive[head] = OrderedDict()
+        self.toc_stack.append(head)
+        self.prev_level = level
+
+    def set_toc(self, _min=1, _max=9, enable_emoji_header=True):
+        self.content.append((_min, _max, enable_emoji_header))
+
+
+class MultilangualDoc(object):
+    class SafetyRead(object):
```

Comment:
Codacy found an issue: [Missing docstring in public nested class (D106)](https://app.codacy.com/gh/mommel/hs-lr-midi-schwurbler/pullRequest?prid=6419091)